internal/netutil/netutil.go

```diff
@@ -0,0 +1,22 @@
+// Package netutil contains various functions that help with networking.
+package netutil
+
+import "net"
+
+// AllocatePorts allocates random ports suitable for listening.
+func AllocatePorts(count int) ([]string, error) {
```

Comment:
My understanding is that `:0` gets a list of system-chosen ports. Does `net.Listen` hold these ports until they are used? 

> If the port in the address parameter is empty or "0", as in "127.0.0.1:" or "[::1]:0", a port number is automatically chosen.

Is there a real risk of another service grabbing one of those ports prior to us using them?

Reply:
It does not hold them. Yes there's a risk of another service grabbing them. I don't know of a way to prevent this.